airbyte-integrations/connectors/source-oracle/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/OracleSourceComprehensiveTest.java

```diff
@@ -0,0 +1,253 @@
+            .addExpectedValues("!\"#$%&'()")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("NVARCHAR2")
+            .airbyteType(JsonSchemaPrimitive.STRING)
+            .fullSourceDataType("NVARCHAR2(3)")
+            .addInsertValues("null", "N'テスト'")
+            .addExpectedValues(null, "テスト")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("NUMBER")
+            .airbyteType(JsonSchemaPrimitive.NUMBER)
+            .addInsertValues("null", "1", "123.45", "power(10, -130)", "9.99999999999999999999 * power(10, 125)")
+            .addExpectedValues(null, "1", "123.45", String.valueOf(Math.pow(10, -130)), String.valueOf(9.99999999999999999999 * Math.pow(10, 125)))
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("NUMBER")
+            .airbyteType(JsonSchemaPrimitive.NUMBER)
+            .fullSourceDataType("NUMBER(6,-2)")
+            .addInsertValues("123.89")
+            .addExpectedValues("100.0")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("FLOAT")
+            .airbyteType(JsonSchemaPrimitive.NUMBER)
+            .fullSourceDataType("FLOAT(5)")
+            .addInsertValues("1.34", "126.45")
+            .addExpectedValues("1.3", "130.0")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("FLOAT")
+            .airbyteType(JsonSchemaPrimitive.NUMBER)
+            .addInsertValues("126.45", "126")
+            .addExpectedValues("126.45", "126")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("BINARY_FLOAT")
+            .airbyteType(JsonSchemaPrimitive.NUMBER)
+            .addInsertValues("126.45f", "1.17549E-38f", "3.40282347E+038f", "BINARY_FLOAT_INFINITY")
+            .addExpectedValues("126.45", "1.17549E-38", "3.4028235E38", "Infinity")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("BINARY_DOUBLE")
+            .airbyteType(JsonSchemaPrimitive.NUMBER)
+            .addInsertValues("126.45d", "2.22507485850720E-308", "TO_BINARY_DOUBLE('1.79769313486231E+308')", "BINARY_DOUBLE_INFINITY")
+            .addExpectedValues("126.45", "0.0", "1.79769313486231E308", "Infinity")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("DATE")
+            .airbyteType(JsonSchemaPrimitive.STRING)
+            .addInsertValues("to_date('-4700/01/01','syyyy/mm/dd')", "to_date('9999/12/31 23:59:59','yyyy/mm/dd hh24:mi:ss')", "null")
+            .addExpectedValues("4700-01-01T00:00:00Z", "9999-12-31T23:59:59Z", null)
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("TIMESTAMP")
```

Comment:
DATETIME, TIMESTAMP WITH LOCAL TIME ZONE, TIMESTAMP WITH TIME ZONE datatype?

Reply:
@yaroslav-hrytsaienko there is no DATETIME datatype. Only DATE, TIMESTAMP WITH LOCAL TIME ZONE, TIMESTAMP WITH TIME ZONE. The last one you can see below. Thanks